src/lib/twitter_web/views/api/post_view.ex

```
@@ -0,0 +1,58 @@
+defmodule TwitterWeb.API.PostView do
+  use TwitterWeb, :view
+
+  @doc """
+  Returns a list of tweets in json form.
+
+  ## Examples
+
+      TwitterWeb.API.PageView.render("index.json", %{tweets: [%Twitter.Tweets.Post{}}]})
+      %{
+        data: %{
+          type: "tweet",
+          content: tweet.content,
+          tweet_id: tweet.id,
+          user: %{
+            type: "user",
+            id: user.id,
+            username: user.username
+          }
+        }
+      }
+  """
+  @spec render(String.t(), map()) :: map()
+  def render("index.json", %{tweets: tweets}) do
+    %{data: render_many(tweets, __MODULE__, "item.json", as: :tweet)}
+  end
+
+  @doc """
+  Returns a one tweet in json form.
+
+  ## Examples
+
+      TwitterWeb.API.PageView.render("index.json", %{tweets: [%Twitter.Tweets.Post{}}]})
+      %{
+          type: "tweet",
+          content: tweet.content,
+          tweet_id: tweet.id,
+          user: %{
+                    type: "user",
+                    id: user.id,
+                    username: user.username
+                  }
+      }
+  """
+  @spec render(String.t(), map()) :: map()
+  def render("item.json", %{tweet: tweet}) do
+    %{
+      type: "tweet",
+      content: tweet.content,
+      tweet_id: tweet.id,
+      user: render_one(tweet.user, TwitterWeb.API.UserView, "user.json", as: :user)
+    }
+  end
+
+  def render(_, _) do
```

is this catch_all necessary? why not let it fail